drivers/tmp006/tmp006.c

```diff
@@ -226,9 +227,26 @@ void tmp006_convert(int16_t rawv, int16_t rawt,  float *tamb, float *tobj)
 
 int tmp006_read_temperature(const tmp006_t *dev, int16_t *ta, int16_t *to)
 {
+    uint8_t drdy;
+
+#if SENSOR_COMPUTOFFLOAD
```

Comment:
where does that come from and what's it doing?

Reply:
I found out that this computation consumes considerable power. So I aim to provide an OPTION that a low power sensing device can send tmp006's raw value to the server so that the server converts the raw value to a readable value.

Reply:
It should be possible to significantly reduce the runtime computations by using uint32_t instead of double, and replace things like pow(x, 2) by x*x etc

Reply:
I can try that. Then pow(x,1/4) would be the only expensive one.

Reply:
You could try to keep it as uint32_t until the final x^4, y^(1/4) and convert to double there. Also, see if sqrt(sqrt(x)) is more efficient than pow(x, .25). A third option is to use an integer square root algorithm, but that may take some effort to ensure that the x^4 does not overflow an uint32_t.

Reply:
Yet another option would be to generate a segmented linearization of the conversion function using look up tables and simple linear interpolation between the points. The accuracy would depend on the size of the segments (which also affect the required ROM for the look up table)

Reply:
We can try to reduce computation power, but whatever we try, I think that computation offloading option is still valuable since it requires ZERO computation. 
FYI, when using 48MHz, the original version requires 1.58ms and using double only for x^4,y^(1/4) requires 0.8ms. x^4 overflows uint32_t. Computation offloading requires 0ms. I think lightweight conversion can be another separate PR. How do you think @gebart ?

Reply:
Yes, let's do the optimizations separately.